### ISSUE TITLE
Add debug option to not remove raw XML attribute (or create a new property and always store it)

### ISSUE THREAD

**Issue by agoode** [openslide/openslide#201]
In Philips (and potentially others), the TIFF description contains an XML file with properties embedded. Currently, OpenSlide parses and discards the XML, but it would be handy to keep around, either with a debug option or a new property.

**Comment by bgilbert** [comment]
For debugging purposes, it would be convenient to extract the XML even if the slide is rejected by all drivers, and even if the slide is not a valid TIFF (so tiffdump may not work, e.g. NDPI > 4 GB).  That leaves either creating a new command-line tool that shares code with tifflike, or adding an OPENSLIDE_DEBUG flag that causes tifflike to dump out all of the ImageDescriptions and XMLPackets.

Did you have a non-debugging purpose in mind?

**Comment by agoode** [comment]
I was only thinking of debugging.

For the Philips case, I needed to get the whole XML (and tiffdump refused to not truncate it). I added the XML data into `philips.XML` (in a local copy of the code), which worked well.

**Comment by bgilbert** [comment]
Okay.  FYI, I believe tiffinfo doesn't truncate.

**Comment by bgilbert** [comment]
I don't think we should encourage applications to parse XML directly, and dump tools are better implemented outside of OpenSlide (`tiffinfo`, a script in `misc`, an ImHex pattern, etc), so I'll close.
